src/main/java/school/hei/haapi/endpoint/rest/controller/FeeController.java

```diff
@@ -161,4 +172,20 @@ public FeeTemplate createOrUpdateFeeTemplate(
     return feeTemplateMapper.toRest(
         feeTemplateService.createOrUpdateFeeTemplate(feeTemplateMapper.toDomain(feeType)));
   }
+
+  @GetMapping(value = "/fees/projection", produces = IMAGE_PNG_VALUE)
+  public ResponseEntity<Resource> getUnpaidFeesProjection() {
+    Instant now = Instant.now();
+    File unpaidFeesGraph = patrimoineService.visualizeUnpaidFees(now);
+
+    if (unpaidFeesGraph == null || !unpaidFeesGraph.exists()) {
+      throw new NotFoundException("Unpaid fees graph not found");
+    }
```

Comment:
should be thrown in the Service, not in the Controller

Reply:
done